dev-client/src/screens/SiteNotesScreen/EditSiteNoteScreen.tsx

```diff
@@ -75,30 +92,35 @@ export const EditSiteNoteScreen = ({note}: Props) => {
 
   const handleDelete = useCallback(async () => {
     setIsSubmitting(true);
-    await dispatch(deleteSiteNote(note)).then(() => navigation.pop());
+    navigation.pop();
+    await dispatch(deleteSiteNote(note));
     setIsSubmitting(false);
   }, [navigation, dispatch, note]);
 
   return (
-    <ScreenFormWrapper
-      ref={formWrapperRef}
-      initialValues={{content: note.content}}
-      onSubmit={handleUpdateNote}
-      onDelete={handleDelete}
-      isSubmitting={isSubmitting}>
-      {formikProps => (
-        <Column pt={10} pl={5} pr={5} pb={10} flex={1}>
-          <Heading variant="h6" pb={7}>
-            {t('site.notes.edit_title')}
-          </Heading>
-          <Box flexGrow={1}>
-            <SiteNoteForm
-              content={formikProps.values.content}
-              editDisabled={!currentUserIsAuthor}
-            />
-          </Box>
-        </Column>
+    <RestrictByRequirements requirements={requirements}>
+      {() => (
+        <ScreenFormWrapper
+          ref={formWrapperRef}
+          initialValues={{content: note.content}}
```

Comment:
I find this ["Function as a child component" pattern](https://reactpatterns.js.org/docs/function-as-child-component/) kind of ugly, but without it, we'd get errors like 
![image](https://github.com/user-attachments/assets/6867ef65-cc3a-4b80-a22b-fdc20e05253e)
because React evaluates it before knowing if its associated component will end up being rendered.

Feel free to give opinions on the options:
- Keep it
- Could instead just change this to `note?.content` and find all the places to put question marks in subsequent screens -- but I feel like this obscures the intent (makes you think that ScreenFormWrapper could render without a note)
- Something else I haven't thought of?

Reply:
We may end up having a pattern where the requirements logic will go in the Screen component, and the rest of the screen's logic will be extracted into a Content component. 
That would at least isolate this to not be mixed in with the screen's content's logic. And maybe we will end up removing RestrictByRequirements if it seems unnecessary after that refactoring.